### PR TITLE
app diagram

### PR DESCRIPTION
**Pull request by sujitn**

### CODE REVIEW

**Review by hugh-emerson — CHANGES_REQUESTED** [review]
First some general tidying up of your repo:

- Why do you still have requirements.txt, requirements-frozen.txt and setup.sh when the app is run with poetry instead?
- You should create directories instead of having all of your files in the root of the project
- You need to update the README. Maintaining good documentation is an important part of making any changes. Could you read that having just freshly cloned the repo and successfully run the app locally? It's missing installing poetry, running `poetry install` or setting up .env. It doesn't mention running tests or any requirements for running tests. It should also mention these diagrams.
- You need to add `.venv` to .gitignore because that's the virtual env directory that gets created when you run `poetry install`.

Actually regarding this PR: the context and container diagrams look good but I think the component one could be improved:

I don't understand what the `@app_route/` label is meant to mean and why it would only apply to POST requests and not GET requests?
In fact I don't think there needs to be any distinction between POST and GET. A component diagram is not a flowchart of application logic - it's an illustration of the building blocks of the app. You could divide them up in the diagram if they were handled by different components in the codebase, but they're all together in app.py

You're going into the level of detail of a Code diagram rather than just component. E.g. By including your python class names and a box for "Form object".

A lot of the boxes are actions rather than components:
- "Submit to Trello" and "load from trello" are not components - they're the interaction between TrelloClient and the external Trello system. There should just be a box for TrelloClient and then an arrow connecting it to the Trello API, which could be labelled with more detail.
- "Display form & table" is not a component. "Templates" would be a component
- "Server side form validation" is not a component. It maybe could be if you had a module which is responsible for server side validation. But actually you don't have any server side validation at all, do you?

**Comment by sujitn** [comment]
Updated diagram, readme and gitignore

File: README.md [hunk 1]
```diff
@@ -24,3 +28,8 @@ You should see output similar to the following:
  * Debugger PIN: 226-556-590
 ```
 Now visit [`http://localhost:5000/`](http://localhost:5000/) in your web browser to view the app.
+
+
+Running the tests
+
+To run all tests, run `pytest`
```
Comment (hugh-emerson) [inline README.md:35]: It's only `pytest` if from a poetry venv shell. Otherwise it needs to be `poetry run pytest`

**Review by hugh-emerson — APPROVED** [review]
Looks a lot better! Merge whenever you're ready 👍

**Pull request merged**